### ISSUE TITLE
Regarding RoundRobin

### ISSUE THREAD

**Issue by JimmySplit** [rofl0r/proxychains-ng#176]
I create a shell script X.sh containing:

curl icanhazip.com
curl icanhazip.com
curl icanhazip.com

then I run:
proxychains4 X.sh

Why round robin won't work in this example? (ofc I uncommented roundrobin from conf)

**Comment by rofl0r** [comment]
dunno, maybe compile with CFLAGS=-DDEBUG ./configure and paste debug output

**Comment by JimmySplit** [comment]
@rofl0r so it's working for you?

**Comment by rofl0r** [comment]
i didnt try; have other things to do

**Comment by rofl0r** [comment]
i merged roundrobin code written by @crass to do him a favor; but he since disappeared. i never used it. if you can hunt down the bug or provide debug info which helps me analyzing it, well; we may be able to fix it. if not, you'll have to live with the bug, i guess.

**Comment by rofl0r** [comment]
now that i think about it, this behaviour seems logical. shell starts a number of independent processes, and each one starts with a fresh round-robin-state.

**Comment by JimmySplit** [comment]
So what's the point of round robin if it won't work with bash?

**Comment by rofl0r** [comment]
i suppose the point is that it's meant to be used for a long-running process, since it cannot work like that with independent processes (unless you start hooking execve too and communicate a global state between all instances, which opens a huge can of worms)

**Comment by JimmySplit** [comment]
Do you have any idea on how one would go about runnig a bash script in the same process?

**Comment by rofl0r** [comment]
yeah. you could put the proxy-selection logic into the bash script itself, and write a different config file for each proxychains invocation.
